dashboard/app/controllers/application_controller.rb

```diff
@@ -271,4 +273,12 @@ def pairing_user_ids
     # TODO(asher): Determine whether we need to guard against it being nil.
     session[:pairings].nil? ? [] : session[:pairings]
   end
+
+  def clear_sign_up_session_vars
+    if session[:sign_up_uid] || session[:sign_up_type] || session[:sign_up_tracking_expiration]
```

Comment:
I think you could remove the `if` statement here - Session.delete is simply Hash.delete, and is safe to call even if the targeted key doesn't exist.

Reply:
That was how I had it initially, but even if the keys aren't there deleting them touches the session, and if it's nil, it appears to create it. That breaks another test (see the last commit in the PR), and I'm not sure how important that is to preserve, so I added this guard.

Reply:
Got it!  I didn't realize that.  TIL.

Reply:
Might be a reasonable simplification then to just say `if session`.

Reply:
Weirdly enough, that doesn't work, either - `session` appears to be truthy, so the deletes still fire.

Reply:
😱 Clearly I need to go re-learn how Rack sessions work.